### PR TITLE
[BI-640] - Disable modification of user role for self

### PR DESCRIPTION
**Pull request by HMS17**
The current messages when a sys admin modifies their own role and no other fields are confusing. To simplify this, the field for modifying a user's role will be disabled when the user is the same as the sys admin.

**Changes:**
When user matches active user:
- In edit for System Users table, disabled Role field
- Only submitted change via promises for user info and not for user roles
- Use more specific user info success message 'User info (name/email/program) successfully updated'

Modified banner message logic for one promise succeeding and not the other to only fire when there are two promises

Linked to PR [taf/BI-640](https://github.com/Breeding-Insight/taf/pull/15)

### CODE REVIEW

File: src/components/admin/AdminUsersTable.vue [hunk 1]
```diff
@@ -442,6 +461,14 @@ export default class AdminUsersTable extends Vue {
     }
   }
 
+  isCurrentUser(userId: string) {
+    if (this.activeUser!.id == userId){
```
Comment (nickpalladino) [inline src/components/admin/AdminUsersTable.vue:465]: Although not as much an issue in Typescript I think we've been sticking with === for equality checks

File: src/components/admin/AdminUsersTable.vue [hunk 2]
```diff
@@ -224,6 +224,7 @@
               v-bind:selected-id="editData.roleId"
               v-bind:field-name="'Role'"
               v-bind:empty-value-name="'No Role'"
+              :isDisabled="isCurrentUser(editData.id)"
```
Comment (nickpalladino) [inline src/components/admin/AdminUsersTable.vue:227]: For consistency I think we've been sticking with the v-bind syntax and kebab case prop name

**Review by nickpalladino — COMMENTED** [review]
Do we need a card for updating TAF test cases?

File: src/components/admin/AdminUsersTable.vue (hunk 1 above)
Comment (timparsons) [reply src/components/admin/AdminUsersTable.vue:465]: +1

**Review by timparsons — APPROVED** [review]
Approving assuming suggested changes from Nick are addressed.

File: src/components/admin/AdminUsersTable.vue (hunk 1 above)
Comment (HMS17) [reply src/components/admin/AdminUsersTable.vue]: Resolved

File: src/components/admin/AdminUsersTable.vue (hunk 2 above)
Comment (HMS17) [reply src/components/admin/AdminUsersTable.vue]: Resolved. Interesting that with vue the @Prop listed can be camel case but the v-bind can have kebab case...

**Comment by HMS17** [comment]
> Do we need a card for updating TAF test cases?

We would, yes. In UserManagementSysAd.feature we'd need possibly one extra scenario to test for for conditional disabling of the Roles field and then modifying BI-836 to no longer change role and not look for the role error banner. Should I make/pick up the card?

**Comment by nickpalladino** [comment]
> > Do we need a card for updating TAF test cases?
> 
> We would, yes. In UserManagementSysAd.feature we'd need possibly one extra scenario to test for for conditional disabling of the Roles field and then modifying BI-836 to no longer change role and not look for the role error banner. Should I make/pick up the card?

Ya, I would say make the card. As far as working on it, not sure what the priorities are, could check with @timparsons

**Review by nickpalladino — APPROVED** [review]

**Comment by HMS17** [comment]
> > > Do we need a card for updating TAF test cases?
> > 
> > 
> > We would, yes. In UserManagementSysAd.feature we'd need possibly one extra scenario to test for for conditional disabling of the Roles field and then modifying BI-836 to no longer change role and not look for the role error banner. Should I make/pick up the card?
> 
> Ya, I would say make the card. As far as working on it, not sure what the priorities are, could check with @timparsons

Added BI-1090. Not sure how high priority it is, but this PR will break scenario BI-836 so it's higher priority than a card that would just consist of adding a new scenario.

**Pull request merged**